demo_nodes_cpp/src/topics/talker_loaned_message.cpp

```diff
@@ -0,0 +1,107 @@
+  : Node("loaned_message_talker", options)
+  {
+    // Create a function for when messages are to be sent.
+    setvbuf(stdout, NULL, _IONBF, BUFSIZ);
+
+    // We differentiate in this demo between two fundamental message types - POD and non-POD
+    // PODs are plain old data types, meaning all the data of its type is encapsulated within
+    // the structure and does not require any heap allocation or dynamic resizing.
+    // non-PODs are essentially the opposite where the data size changes during runtime.
+    // All containers (including Strings) are such non-PODs.
+    // Most middlewares won't be able to loan non-POD datatypes.
+    // We thus feature two publishers in this demo where both, a POD and non-POD message
+    // will be used to publish data.
+    // The take-away for this is that the rclcpp API for message loaning can cope with
+    // either POD and non-POD transparently.
+    auto publish_message =
+      [this]() -> void
+      {
+        // We loan a message here and don't allocate the memory on the stack.
+        // For middlewares which support message loaning, this means the middleware
+        // completely owns the memory for this message.
+        // This enables a zero-copy message transport for middlewares with shared memory
+        // capabilities.
+        // If the middleware doesn't support this, the loaned message will be allocated
+        // with the allocator instance provided by the publisher.
+        auto pod_loaned_msg = pod_pub_->loan_message();
+        auto pod_msg_data = static_cast<double>(count_++);
+        pod_loaned_msg.get().data = pod_msg_data;
+        RCLCPP_INFO(this->get_logger(), "Publishing: '%f'", pod_msg_data);
+        // As the middleware might own the memory allocated for this message,
+        // a call to publish explicitly transfers ownership back to the middleware.
+        // The loaned message instance is thus no longer valid after a call to publish.
+        pod_pub_->publish(std::move(pod_loaned_msg));
+
+        // Similar as in the above case, we ask the middleware to laon a message.
+        // As most likely the middleware won't be able to loan a message for a non-POD
+        // data type, the memory for the message will be allocated on the heap within
+        // the scope of the `LoanedMessage` instance.
+        // After the call to `publish()`, the message will be correctly allocated.
+        auto non_pod_loaned_msg = non_pod_pub_->loan_message();
+        auto non_pod_msg_data = "Hello World: " + std::to_string(count_++);
+        non_pod_loaned_msg.get().data = non_pod_msg_data;
+        RCLCPP_INFO(this->get_logger(), "Publishing: '%s'", non_pod_msg_data.c_str());
+        non_pod_pub_->publish(std::move(non_pod_loaned_msg));
+      };
+
+    // Create a publisher with a custom Quality of Service profile.
+    rclcpp::QoS qos(rclcpp::KeepLast(7));
+    pod_pub_ = this->create_publisher<std_msgs::msg::Float64>("/chatter_pod", qos);
```

Comment:
Is it correct to make this a globally namespaced topic? Then if the node is namespaced, the topic will not change. Just checking if we want this behavior or if this should be changed to `"chatter_pod"`. Same for the topic.